internal/convenience/repository/synchronizer_repository.go

```diff
@@ -51,17 +56,22 @@ func (c *SynchronizerRepository) Create(
 		ini_report_cursor_after,
 		end_report_cursor_after
 		) VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`
-	c.Db.MustExec(
-		insertSql,
+
+	exec := DBExecutor{&c.Db}
+	_, err := exec.ExecContext(ctx, insertSql,
 		data.TimestampAfter,
 		data.IniCursorAfter,
 		data.LogVouchersIds,
 		data.EndCursorAfter,
 		data.IniInputCursorAfter,
 		data.EndInputCursorAfter,
 		data.IniReportCursorAfter,
-		data.EndReportCursorAfter,
-	)
+		data.EndReportCursorAfter)
+
+	if err != nil {
+		return nil, err
+	}
+
 	return data, nil
```

Comment:
You can return
```go
return data, err
```
instead.

Reply:
Why should I return data if an error occurs?

Reply:
Because in outside pattern, you always check error first. So, you don't need worry about it.